
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
        <link rel="canonical" href="https://myloasm-docs.github.io/output/">
      
      
        <link rel="prev" href="../cmd/">
      
      
        <link rel="next" href="../secondary/">
      
      
      <link rel="icon" href="../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.6.20">
    
    
      
        <title>Primary files - myloasm - metagenomic assembly with (noisy) long reads</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/main.e53b48f4.min.css">
      
        
        <link rel="stylesheet" href="../assets/stylesheets/palette.06af60db.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
    <script>__md_scope=new URL("..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="teal" data-md-color-accent="indigo">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#output-directory" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--shadow" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href=".." title="myloasm - metagenomic assembly with (noisy) long reads" class="md-header__button md-logo" aria-label="myloasm - metagenomic assembly with (noisy) long reads" data-md-component="logo">
      
  <img src="../assets/logo.svg" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            myloasm - metagenomic assembly with (noisy) long reads
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Primary files
            
          </span>
        </div>
      </div>
    </div>
    
      
    
    
    
    
      
      
        <label class="md-header__button md-icon" for="__search">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        </label>
        <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
      
    
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    



<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href=".." title="myloasm - metagenomic assembly with (noisy) long reads" class="md-nav__button md-logo" aria-label="myloasm - metagenomic assembly with (noisy) long reads" data-md-component="logo">
      
  <img src="../assets/logo.svg" alt="logo">

    </a>
    myloasm - metagenomic assembly with (noisy) long reads
  </label>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href=".." class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Introduction
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../install/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Install
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../usage/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Quick usage
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../cmd/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Full option reference
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
    
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5" checked>
        
          
          <label class="md-nav__link" for="__nav_5" id="__nav_5_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Output
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_5_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_5">
            <span class="md-nav__icon md-icon"></span>
            Output
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  
  <span class="md-ellipsis">
    Primary files
    
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  
  <span class="md-ellipsis">
    Primary files
    
  </span>
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#the-main-contig-file-assembly_primaryfa" class="md-nav__link">
    <span class="md-ellipsis">
      The main contig file: assembly_primary.fa
    </span>
  </a>
  
    <nav class="md-nav" aria-label="The main contig file: assembly_primary.fa">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#contig-fasta-record-label-definition" class="md-nav__link">
    <span class="md-ellipsis">
      Contig fasta record label definition
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Contig fasta record label definition">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#circularity" class="md-nav__link">
    <span class="md-ellipsis">
      Circularity
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#estimated-depth-of-coverage" class="md-nav__link">
    <span class="md-ellipsis">
      Estimated depth of coverage
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#k-mer-multiplicity-and-duplication" class="md-nav__link">
    <span class="md-ellipsis">
      K-mer multiplicity and duplication
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#the-final-assembly-graph-final_contig_graphgfa" class="md-nav__link">
    <span class="md-ellipsis">
      The final assembly graph: final_contig_graph.gfa
    </span>
  </a>
  
    <nav class="md-nav" aria-label="The final assembly graph: final_contig_graph.gfa">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#understanding-myloasms-gfa-assembly-graph-format" class="md-nav__link">
    <span class="md-ellipsis">
      Understanding myloasm's GFA assembly graph format
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Understanding myloasm&#39;s GFA assembly graph format">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#contig-information" class="md-nav__link">
    <span class="md-ellipsis">
      Contig information
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#read-information" class="md-nav__link">
    <span class="md-ellipsis">
      Read information:
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../secondary/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Secondary files
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_6" >
        
          
          <label class="md-nav__link" for="__nav_6" id="__nav_6_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Quality control tools
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_6_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_6">
            <span class="md-nav__icon md-icon"></span>
            Quality control tools
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../qc/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    QC Overview
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../qc_statistics/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    QC Statistics
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../qc_manual/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Manual inspection with mylotools
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../results/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Preliminary Results
    
  </span>
  

      </a>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#the-main-contig-file-assembly_primaryfa" class="md-nav__link">
    <span class="md-ellipsis">
      The main contig file: assembly_primary.fa
    </span>
  </a>
  
    <nav class="md-nav" aria-label="The main contig file: assembly_primary.fa">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#contig-fasta-record-label-definition" class="md-nav__link">
    <span class="md-ellipsis">
      Contig fasta record label definition
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Contig fasta record label definition">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#circularity" class="md-nav__link">
    <span class="md-ellipsis">
      Circularity
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#estimated-depth-of-coverage" class="md-nav__link">
    <span class="md-ellipsis">
      Estimated depth of coverage
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#k-mer-multiplicity-and-duplication" class="md-nav__link">
    <span class="md-ellipsis">
      K-mer multiplicity and duplication
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#the-final-assembly-graph-final_contig_graphgfa" class="md-nav__link">
    <span class="md-ellipsis">
      The final assembly graph: final_contig_graph.gfa
    </span>
  </a>
  
    <nav class="md-nav" aria-label="The final assembly graph: final_contig_graph.gfa">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#understanding-myloasms-gfa-assembly-graph-format" class="md-nav__link">
    <span class="md-ellipsis">
      Understanding myloasm's GFA assembly graph format
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Understanding myloasm&#39;s GFA assembly graph format">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#contig-information" class="md-nav__link">
    <span class="md-ellipsis">
      Contig information
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#read-information" class="md-nav__link">
    <span class="md-ellipsis">
      Read information:
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  



<h1 id="output-directory">Output directory</h1>
<div class="highlight"><pre><span></span><code>myloasm_output 
    ├── assembly_primary.fa &lt;-- CONTIGS
    ├── final_contig_graph.(gfa/edges) &lt;-- GRAPH 
    └── OTHER STUFF (secondary files)
</code></pre></div>
<ul>
<li><a href="#the-main-contig-file-assembly_primaryfa">The main contig file: <strong>assembly_primary.fa</strong></a><ul>
<li><a href="#contig-fasta-record-label-definition">Contig fasta record label definition</a><ul>
<li><a href="#circularity">Circularity</a></li>
<li><a href="#estimated-depth-of-coverage">Estimated depth of coverage </a></li>
<li><a href="#k-mer-multiplicity">K-mer multiplicity</a></li>
</ul>
</li>
</ul>
</li>
<li><a href="#the-final-assembly-graph-final_contig_graphgfa">The final assembly graph: <strong>final_contig_graph.gfa</strong></a><ul>
<li><a href="#understanding-myloasms-gfa-assembly-graph-format">Understanding myloasm's GFA assembly graph format </a><ul>
<li><a href="#contig-information">Contig information </a></li>
<li><a href="#read-information">Read information </a></li>
</ul>
</li>
</ul>
</li>
</ul>
<h2 id="the-main-contig-file-assembly_primaryfa">The main contig file: <strong>assembly_primary.fa</strong></h2>
<p>These are the main polished contigs to be used for downstream analysis most of the time. </p>
<h3 id="contig-fasta-record-label-definition">Contig fasta record label definition</h3>
<p>The fasta records looks like this:</p>
<blockquote>
<p>&gt;u2658272ctg_len-2275750_circular-possibly_depth-10-9-9_duplicated-no mult=1.00</p>
</blockquote>
<p><code>u2658272ctg</code> is the identifier and <code>len-X</code> is the length in nucleotides. There is also information about the circularity, depth of coverage, and repetitiveness of the contig. </p>
<h4 id="circularity">Circularity</h4>
<p><code>circular-X</code>: X is either <code>yes</code>, <code>no</code>, or <code>possibly</code>. </p>
<p>If a contig has a self loop in the assembly graph and </p>
<ol>
<li>has average depth &gt; 5x and </li>
<li>has no other connections (i.e., consists of a single node and a single edge) </li>
</ol>
<p>then it is <code>yes</code>. </p>
<p>If it has a self loop, but (1) or (2) fails above, then it is <code>possibly</code>. Otherwise, it is <code>no</code>. </p>
<h4 id="estimated-depth-of-coverage">Estimated depth of coverage</h4>
<p><code>depth-X1-X2-X3</code>: This is the estimated depth of coverage.</p>
<p>The three distinct values represent different levels of nucleotide identity thresholds. </p>
<ol>
<li><code>X1</code> is the depth while allowing alignments of approximately &gt; 99% true nucleotide similarity (estimated using myloasm's SNP + k-mer formulation). </li>
<li><code>X2</code> is the depth for &gt; 99.75% similarity.</li>
<li><code>X3</code> is the depth for 100% similarity.</li>
</ol>
<h4 id="k-mer-multiplicity-and-duplication">K-mer multiplicity and duplication</h4>
<p><code>mult=X</code>: this is the estimated fraction of repetitiveness and useful for quality control. </p>
<p>This is estimated by counting 21-mers across the contig and seeing how often they repeat on average after removing the most frequent and least frequent 10% of k-mers. </p>
<ul>
<li>Most complete prokaryotic genomes should have <code>mult-1.00</code>. </li>
<li>Sometimes myloasm erroneously concatenates two similar strain genomes together. In this case, <code>mult</code> is &gt; 1.0 and indicative of contamination. </li>
<li>Long reads often contain duplicate plasmids. Sometimes, a plasmid is erroneously duplicated or miscircularized, giving <code>mult</code> &gt; 1. </li>
</ul>
<p>If <code>mult</code> is &gt; 1.1, then <code>duplicated</code> is <code>possibly</code>. If &gt; 1.5, then <code>yes</code>. Otherwise, it is <code>no</code>. </p>
<h2 id="the-final-assembly-graph-final_contig_graphgfa">The final assembly graph: <strong>final_contig_graph.gfa</strong></h2>
<p>Each contig in <code>assembly_primary.fa</code> is a node in the final assembly graph, <code>final_contig_graph.gfa</code>. This is in modified <a href="https://gfa-spec.github.io/GFA-spec/GFA1.html">GFA format</a> and can be visualized with <a href="https://rrwick.github.io/Bandage/">Bandage</a>.</p>
<p>Each <code>*.gfa</code> file has a corresponding <code>*.edges</code> file. The edges files gives more detailed information about contig-to-contig overlaps. </p>
<h3 id="understanding-myloasms-gfa-assembly-graph-format">Understanding myloasm's GFA assembly graph format</h3>
<p>Myloasm's GFA file contains a header starting with <code>H</code>. Then, there is additional contig + read information that may be useful for manual curation.  </p>
<h4 id="contig-information">Contig information</h4>
<div class="highlight"><pre><span></span><code>S       u165592ctg              LN:i:1034269    DP:f:25.0       DP2:f:24.0      DP3:f:23.0      MEDIAN_DP1:f:35.0
</code></pre></div>
<ul>
<li><strong>NOTE</strong>: <code>LN:i</code> is only an <em>approximate</em> length. This is the length <strong>before polishing</strong>.</li>
<li><code>DP</code>, <code>DP2</code>, and <code>DP3</code> is the median of all <code>DP/2/3</code> values across the reads (see below). <a href="#estimated-depth-of-coverage">See here for more information on depth</a>. </li>
<li><code>MEDIAN_DP1</code> is the median-of-the-median-depth across all reads. <code>DP</code> is the median-of-the-<em>minimum</em>-depth across all reads. You can ignore the difference and use <code>DP</code> instead of <code>MEDIAN_DP1</code> most of the time. </li>
</ul>
<h4 id="read-information">Read information:</h4>
<div class="highlight"><pre><span></span><code>a       u165592ctg,0-28710      165592  0       SRRX.2+split0       -       28710   DP1:22,DP2:22,DP3:21,READ_LEN:43601,OL_LEN_NEXT:14808,SNP_SHARE_NEXT:216,SNP_DIFF_NEXT:1
a       u165592ctg,34-6529      1421634 28710   SRRX.1+split0+split0        +       6495    DP1:42,DP2:42,DP3:13,READ_LEN:21212,OL_LEN_NEXT:14629,SNP_SHARE_NEXT:129,SNP_DIFF_NEXT:0
</code></pre></div>
<ol>
<li><code>a</code> indicates this is a read</li>
<li><code>u165592ctg,0-28710</code> indicates that positions 0-28710 come from read <code>SRRX.2+split0</code> <strong>before polishing</strong>.</li>
<li><code>165592</code> is the numeric read id. Myloasm assigns each read as a unique id. </li>
<li><code>0</code> indicates this read's contribution to the unpolished contig starts at position 0</li>
<li><code>SRRX.2+split0</code> - Read name. If myloasm thinks a read is chimeric, it may be split into several sub reads. For example, <code>split1</code> indicates the second chunk after splitting. Unsplit reads may also be denoted as <code>+split0</code>. There are two rounds of chimera splitting. </li>
<li><code>-/+</code> - read orientation</li>
</ol>
<p>The last columns indicate the depth and overlap information of each read. <a href="#estimated-depth-of-coverage">See here for more information on varying depth values</a>. </p>
<ul>
<li><code>OL_LEN_NEXT</code> - number of overlapping bases to the next read</li>
<li><code>SNP_SHARE_NEXT</code> - number of estimated shared SNPs to the next read</li>
<li><code>SNP_DIFF_NEXT</code> - number of estimated different SNPs to the next read. </li>
</ul>












                
              </article>
            </div>
          
          
<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    
      
      <script id="__config" type="application/json">{"base": "..", "features": [], "search": "../assets/javascripts/workers/search.973d3a69.min.js", "tags": null, "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}, "version": null}</script>
    
    
      <script src="../assets/javascripts/bundle.f55a23d4.min.js"></script>
      
    
  </body>
</html>