
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
        <link rel="canonical" href="https://myloasm-docs.github.io/qc_manual/">
      
      
        <link rel="prev" href="../qc_statistics/">
      
      
        <link rel="next" href="../results/">
      
      
      <link rel="icon" href="../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.6.20">
    
    
      
        <title>Manual inspection with mylotools - myloasm - metagenomic assembly with (noisy) long reads</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/main.e53b48f4.min.css">
      
        
        <link rel="stylesheet" href="../assets/stylesheets/palette.06af60db.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
    <script>__md_scope=new URL("..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="teal" data-md-color-accent="indigo">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#using-mylotools-plot-to-manually-inspect-long-read-overlaps-and-find-chimeras" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--shadow" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href=".." title="myloasm - metagenomic assembly with (noisy) long reads" class="md-header__button md-logo" aria-label="myloasm - metagenomic assembly with (noisy) long reads" data-md-component="logo">
      
  <img src="../assets/logo.svg" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            myloasm - metagenomic assembly with (noisy) long reads
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Manual inspection with mylotools
            
          </span>
        </div>
      </div>
    </div>
    
      
    
    
    
    
      
      
        <label class="md-header__button md-icon" for="__search">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        </label>
        <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
      
    
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    



<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href=".." title="myloasm - metagenomic assembly with (noisy) long reads" class="md-nav__button md-logo" aria-label="myloasm - metagenomic assembly with (noisy) long reads" data-md-component="logo">
      
  <img src="../assets/logo.svg" alt="logo">

    </a>
    myloasm - metagenomic assembly with (noisy) long reads
  </label>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href=".." class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Introduction
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../install/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Install
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../usage/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Quick usage
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../cmd/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Full option reference
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5" >
        
          
          <label class="md-nav__link" for="__nav_5" id="__nav_5_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Output
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_5_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_5">
            <span class="md-nav__icon md-icon"></span>
            Output
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../output/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Primary files
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../secondary/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Secondary files
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
    
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_6" checked>
        
          
          <label class="md-nav__link" for="__nav_6" id="__nav_6_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Quality control tools
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_6_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_6">
            <span class="md-nav__icon md-icon"></span>
            Quality control tools
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../qc/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    QC Overview
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../qc_statistics/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    QC Statistics
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  
  <span class="md-ellipsis">
    Manual inspection with mylotools
    
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  
  <span class="md-ellipsis">
    Manual inspection with mylotools
    
  </span>
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#using-mylotools-plot-to-manually-inspect-long-read-overlaps-and-find-chimeras" class="md-nav__link">
    <span class="md-ellipsis">
      Using mylotools plot to manually inspect long-read overlaps and find chimeras
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Using mylotools plot to manually inspect long-read overlaps and find chimeras">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#mylotools-plot-output" class="md-nav__link">
    <span class="md-ellipsis">
      mylotools plot output
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#panel-1-gc-content" class="md-nav__link">
    <span class="md-ellipsis">
      Panel 1: GC content
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#panel-2-cumulative-gc-skew" class="md-nav__link">
    <span class="md-ellipsis">
      Panel 2: Cumulative GC Skew
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#panel-3-coverage-across-the-genome" class="md-nav__link">
    <span class="md-ellipsis">
      Panel 3: Coverage across the genome
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#panel-4-read-overlap-information-within-assembly-graph" class="md-nav__link">
    <span class="md-ellipsis">
      Panel 4: Read overlap information within assembly graph
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#finding-chimeric-breakpoints-from-the-plot-graph" class="md-nav__link">
    <span class="md-ellipsis">
      Finding chimeric breakpoints from the plot graph
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Finding chimeric breakpoints from the plot graph">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#case-study-the-above-graph" class="md-nav__link">
    <span class="md-ellipsis">
      Case study: the above graph
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../results/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Preliminary Results
    
  </span>
  

      </a>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#using-mylotools-plot-to-manually-inspect-long-read-overlaps-and-find-chimeras" class="md-nav__link">
    <span class="md-ellipsis">
      Using mylotools plot to manually inspect long-read overlaps and find chimeras
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Using mylotools plot to manually inspect long-read overlaps and find chimeras">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#mylotools-plot-output" class="md-nav__link">
    <span class="md-ellipsis">
      mylotools plot output
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#panel-1-gc-content" class="md-nav__link">
    <span class="md-ellipsis">
      Panel 1: GC content
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#panel-2-cumulative-gc-skew" class="md-nav__link">
    <span class="md-ellipsis">
      Panel 2: Cumulative GC Skew
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#panel-3-coverage-across-the-genome" class="md-nav__link">
    <span class="md-ellipsis">
      Panel 3: Coverage across the genome
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#panel-4-read-overlap-information-within-assembly-graph" class="md-nav__link">
    <span class="md-ellipsis">
      Panel 4: Read overlap information within assembly graph
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#finding-chimeric-breakpoints-from-the-plot-graph" class="md-nav__link">
    <span class="md-ellipsis">
      Finding chimeric breakpoints from the plot graph
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Finding chimeric breakpoints from the plot graph">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#case-study-the-above-graph" class="md-nav__link">
    <span class="md-ellipsis">
      Case study: the above graph
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  



  <h1>Manual inspection with mylotools</h1>

<div class="admonition important">
<p class="admonition-title">Important</p>
<p>We will be using <a href="../qc/#mylotools---scripts-for-myloasm-outputs">mylotools</a>, a set of scripts for manipulating myloasm's outputs. </p>
</div>
<h2 id="using-mylotools-plot-to-manually-inspect-long-read-overlaps-and-find-chimeras">Using <code>mylotools plot</code> to manually inspect long-read overlaps and find chimeras</h2>
<p>We provide a plotting utility command called <code>mylotools plot</code> to inspect GC content, read overlaps, and coverage information across a contig. Given a contig of interest: e.g. <code>uXXXXXctg</code>, simply do</p>
<div class="highlight"><pre><span></span><code><span class="c1"># you should be in the results directory</span>
<span class="nb">cd</span><span class="w"> </span>myloasm_results<span class="w"> </span>

mylotools<span class="w"> </span>plot<span class="w"> </span>u2912759ctg
ls<span class="w"> </span>u2912759ctg_analysis.html
</code></pre></div>
<h3 id="mylotools-plot-output"><code>mylotools plot</code> output</h3>
<p>Open the file <code>u2912759ctg_analysis.html</code> in a browser and it will look like:</p>
<p><img alt="myloplot output" src="../assets/myloplot2.png" /></p>
<h3 id="panel-1-gc-content">Panel 1: GC content</h3>
<ul>
<li>The first panel is the GC content averaged across windows. </li>
<li>Prokaryotic genomes should have relatively consistent GC content across the genome. If you see a sharp change in GC content, it may indicate a chimeric breakpoint. </li>
</ul>
<h3 id="panel-2-cumulative-gc-skew">Panel 2: Cumulative GC Skew</h3>
<ul>
<li>See <a href="https://pmc.ncbi.nlm.nih.gov/articles/PMC5538567/">this article</a> or <a href="https://academic.oup.com/nar/article/26/10/2286/1030593">this one</a> for more information about GC skew.</li>
<li>Many prokaryotes have a peak and a trough, indicating origin and terminus of replication. </li>
</ul>
<h3 id="panel-3-coverage-across-the-genome">Panel 3: Coverage across the genome</h3>
<ul>
<li>Each dot is a read that the contig was constructed from.</li>
<li>There are 3 different read coverage values. <a href="../output/#estimated-depth-of-coverage">See here</a> for more information about what they mean. DP1 is the most permissive coverage value whereas DP3 indicates exact coverage (perfect alignments). </li>
</ul>
<h3 id="panel-4-read-overlap-information-within-assembly-graph">Panel 4: Read overlap information within assembly graph</h3>
<ul>
<li>Each dot is a read overlap. Statistics about lengths/the read IDs are shown.</li>
<li>The colour indicates the number of shared/differing SNPs between the reads. An ideal contig involves no differing SNPs between reads, but sometimes this is impossible (e.g. high levels of recombination). </li>
</ul>
<h2 id="finding-chimeric-breakpoints-from-the-plot-graph">Finding chimeric breakpoints from the <code>plot</code> graph</h2>
<p>The x-axis are <em>roughly</em> shared across all of the four plots. The first two panels x-axis are for the polished contig. The bottom two are the raw, unpolished contig. These differ very slightly. </p>
<p>To find breakpoints, we can look at points on the x-axis where there are unexpected changes.</p>
<ol>
<li>GC content should be <em>roughly</em> constant across a prokaryotic genome</li>
<li>GC skew often has a single peak and a trough (<a href="https://academic.oup.com/nar/article/26/10/2286/1030593">see here</a>)</li>
<li>Coverage should be constant across a contig. However, DP1 (green) and DP2 (yellow) can vary significantly -- inexact repeat regions (across strains or species) can elevate the DP1/DP2 values.</li>
<li>Read overlaps should be <em>long</em>. Small overlaps can be suspicious. High # of SNP differences are not ideal, but sometimes can not be avoided. </li>
</ol>
<h3 id="case-study-the-above-graph">Case study: the above graph</h3>
<p>The <a href="#mylotools-plot-output">above graph</a> is for a 99% complete but <strong>18% contaminated</strong> contig from an anaerobic digester metagenome. We see irregularities at around the 2.6 Mbp position.</p>
<p><img alt="myloplot bp" src="../assets/myloplot-bp.png" /></p>
<p>A few issues occur:</p>
<ol>
<li>The GC content has a noticeable downward shift</li>
<li>The cumulative GC skew goes up and down again</li>
<li>The DP1/DP2 coverages are <em>variable and high</em>, whereas DP1/DP2 before the breakpoint are <em>consistently low</em></li>
</ol>
<p>This is quite strong evidence for a chimeric join. After removing the bases from 2.6 Mbp onward and rerunning CheckM2, we get a 97% complete and <strong>0% contaminated</strong> contig, confirming our suspicion. </p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Around the breakpoint, all overlaps have length &gt; 8kbp and are <em>almost perfect</em>. This was probably a recent horizontally transferred region. Even with long reads, metagenomic assembly is still extremely difficult. </p>
</div>












                
              </article>
            </div>
          
          
<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    
      
      <script id="__config" type="application/json">{"base": "..", "features": [], "search": "../assets/javascripts/workers/search.973d3a69.min.js", "tags": null, "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}, "version": null}</script>
    
    
      <script src="../assets/javascripts/bundle.f55a23d4.min.js"></script>
      
    
  </body>
</html>