
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
        <link rel="canonical" href="https://myloasm-docs.github.io/qc/">
      
      
        <link rel="prev" href="../mylotools/">
      
      
        <link rel="next" href="../qc_manual/">
      
      
        
      
      
      <link rel="icon" href="../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.7.1">
    
    
      
        <title>QC discussion - myloasm - metagenomic assembly with (noisy) long reads</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/main.484c7ddc.min.css">
      
        
        <link rel="stylesheet" href="../assets/stylesheets/palette.ab4e12ef.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
    <script>__md_scope=new URL("..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
  </head>
  
  
    
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="teal" data-md-color-accent="indigo">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#notes-on-misassemblies" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--shadow" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href=".." title="myloasm - metagenomic assembly with (noisy) long reads" class="md-header__button md-logo" aria-label="myloasm - metagenomic assembly with (noisy) long reads" data-md-component="logo">
      
  <img src="../assets/logo.svg" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            myloasm - metagenomic assembly with (noisy) long reads
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              QC discussion
            
          </span>
        </div>
      </div>
    </div>
    
      
    
    
    
    
      
      
        <label class="md-header__button md-icon" for="__search">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        </label>
        <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
      
    
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    



<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href=".." title="myloasm - metagenomic assembly with (noisy) long reads" class="md-nav__button md-logo" aria-label="myloasm - metagenomic assembly with (noisy) long reads" data-md-component="logo">
      
  <img src="../assets/logo.svg" alt="logo">

    </a>
    myloasm - metagenomic assembly with (noisy) long reads
  </label>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href=".." class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Introduction
  

    
  </span>
  
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../install/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Install
  

    
  </span>
  
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../usage/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Quick usage
  

    
  </span>
  
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4" >
        
          
          <label class="md-nav__link" for="__nav_4" id="__nav_4_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    
  
    Output
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_4_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_4">
            <span class="md-nav__icon md-icon"></span>
            
  
    Output
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../output/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Primary files
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../secondary/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Secondary files
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
    
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5" checked>
        
          
          <label class="md-nav__link" for="__nav_5" id="__nav_5_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    
  
    Mylotools - quality control and manipulation
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_5_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_5">
            <span class="md-nav__icon md-icon"></span>
            
  
    Mylotools - quality control and manipulation
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../mylotools/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Mylotools overview
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  
  <span class="md-ellipsis">
    
  
    QC discussion
  

    
  </span>
  
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  
  <span class="md-ellipsis">
    
  
    QC discussion
  

    
  </span>
  
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#notes-on-misassemblies" class="md-nav__link">
    <span class="md-ellipsis">
      
        Notes on misassemblies
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Notes on misassemblies">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#using-checkm2-on-long-prokaryotic-contigs" class="md-nav__link">
    <span class="md-ellipsis">
      
        Using CheckM2 on long prokaryotic contigs
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#using-k-mer-multiplicity-statistics-for-duplicatesstrain-chimeras" class="md-nav__link">
    <span class="md-ellipsis">
      
        Using k-mer multiplicity statistics for duplicates/strain chimeras
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#notes-on-circularized-contigs" class="md-nav__link">
    <span class="md-ellipsis">
      
        Notes on circularized contigs
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../qc_manual/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Manual inspection with mylotools
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../results/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Preliminary Results
  

    
  </span>
  
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_7" >
        
          
          <label class="md-nav__link" for="__nav_7" id="__nav_7_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    
  
    Advanced info
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_7_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_7">
            <span class="md-nav__icon md-icon"></span>
            
  
    Advanced info
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../cmd/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Full option reference
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../CHANGELOG/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Changelog
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../FAQ/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Frequently asked questions
  

    
  </span>
  
  

      </a>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#notes-on-misassemblies" class="md-nav__link">
    <span class="md-ellipsis">
      
        Notes on misassemblies
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Notes on misassemblies">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#using-checkm2-on-long-prokaryotic-contigs" class="md-nav__link">
    <span class="md-ellipsis">
      
        Using CheckM2 on long prokaryotic contigs
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#using-k-mer-multiplicity-statistics-for-duplicatesstrain-chimeras" class="md-nav__link">
    <span class="md-ellipsis">
      
        Using k-mer multiplicity statistics for duplicates/strain chimeras
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#notes-on-circularized-contigs" class="md-nav__link">
    <span class="md-ellipsis">
      
        Notes on circularized contigs
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              
              <article class="md-content__inner md-typeset">
                
                  



  <h1>QC discussion</h1>

<h2 id="notes-on-misassemblies">Notes on misassemblies</h2>
<p>Like all other metagenome assemblers, myloasm can produce erroneous assemblies.</p>
<p><strong>Chimeric contigs:</strong> chimeric contigs consist of sequences from two or more genomes. This error can happen for a variety of reasons. The most common are:</p>
<ul>
<li>long interspecies shared genomic regions (e.g., horizontal gene transfer) can confuse assemblers</li>
<li>extremely similar strains are hard to resolve, leading to strain-chimeric contigs</li>
<li>chimeric long-reads due to technical sequencing artifacts can cause downstream errors in assembly</li>
</ul>
<p><strong>Duplicated small contigs:</strong> we noticed that long-read assemblers can incorrectly duplicate small sequences (plasmids or viruses). </p>
<p>The cause isn't always obvious, but we often found that a <em>single long read</em> can consist of <em>multiple copies</em> of the same plasmid. </p>
<p><strong>Prematurely circularized contigs:</strong> assemblers can prematurely circularize contigs, i.e., claim contigs are circular even though they are not. Algorithmically, this arises due to genome repeats.</p>
<h3 id="using-checkm2-on-long-prokaryotic-contigs">Using CheckM2 on long prokaryotic contigs</h3>
<p>The easiest way to filter out long (&gt;300 kb) erroneous prokaryotic contigs is to run <a href="https://github.com/chklovski/CheckM2">CheckM2</a> on individual contigs. </p>
<p>If a chimeric contig is long enough, it will often have high CheckM contamination due to the presence of duplicated marker genes from multiple organisms. </p>
<p>To extract all contigs of length &gt;= X bp and run CheckM2, you can do (with <a href="../mylotools/">mylotools</a>)</p>
<div class="highlight"><pre><span></span><code><span class="nb">cd</span><span class="w"> </span>myloasm_results
mylotools<span class="w"> </span>extract-contigs<span class="w"> </span>--output-folder<span class="w"> </span>contigs_dir<span class="w"> </span>--min-contig-length<span class="w"> </span>X
checkm2<span class="w"> </span>predict<span class="w"> </span>--input<span class="w"> </span>contigs_dir<span class="w"> </span>-x<span class="w"> </span>fa<span class="w"> </span>-o<span class="w"> </span>checkm2_results<span class="w"> </span>--threads<span class="w"> </span><span class="m">40</span>
</code></pre></div>
<h3 id="using-k-mer-multiplicity-statistics-for-duplicatesstrain-chimeras">Using k-mer multiplicity statistics for duplicates/strain chimeras</h3>
<p>Myloasm's <a href="../output/">fasta outputs</a> have information about how often 21-mers are repeated (its multiplicity) within a contig: </p>
<blockquote>
<p>&gt;u123123ctg_XXX_<strong>duplicated-yes mult=2.00</strong> &lt;- fasta record with k-mer multiplicity and duplication status.</p>
</blockquote>
<p>In our experience, <em>prokaryotic</em> contigs should almost always have average k-mer multiplicity near 1.00. If you have a very long contig (&gt; 1M bp) of multiplicity &gt; 1.05, it may be a chimera from multiple strains of a species. We set <code>duplicated</code> to <code>yes</code> or <code>possibly</code> then the multiplicity is high. </p>
<p><strong>For small genomes (e.g. viruses)</strong>, the expected k-mer multiplicity may deviate from 1.00. However, a small contig with k-mer multiplicity &gt;&gt; 1 can be suspicious. A contig with k-mer multiplicity = 2, 3, or an integer multiple can indicate a perfectly duplicated contig. </p>
<h3 id="notes-on-circularized-contigs">Notes on circularized contigs</h3>
<p>For prokaryotic genomes, low CheckM2 completeness can indicate premature circularization. However: </p>
<ul>
<li>we have found that complete <em>organelle</em> genomes (mitochondria, plastids from microeukaryotes) can have non-trivial CheckM2 completeness (&gt; 30% but &lt; 90%) </li>
<li><em>secondary chromosomes</em> and genomes with multiple chromosomes can have lower completeness</li>
<li>some clades of microbes have low CheckM2 completeness scores, even when they're complete</li>
</ul>
<p>The myloasm tag <code>circular-possibly</code> indicates lower confidence circular genomes (due to low coverage or assembly graph ambiguity), but these can often be complete genomes, especially if CheckM2 scores are good. </p>












                
              </article>
            </div>
          
          
<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    
      
      
      <script id="__config" type="application/json">{"annotate": null, "base": "..", "features": [], "search": "../assets/javascripts/workers/search.2c215733.min.js", "tags": null, "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}, "version": null}</script>
    
    
      <script src="../assets/javascripts/bundle.79ae519e.min.js"></script>
      
    
  </body>
</html>